### DISCUSSION TITLE
Playlist issues

### DISCUSSION THREAD

**Discussion by AndyFx12** [rern/rAudio#1834] [Bug Reports]
I've noticed some playlist issues:

Steps to reproduce:
- have some songs in your playlist (in my case: 70 songs) and reload the page
- change tab to playlist
- tap e. g. 5th song
**-> song begins to play, counter runs to fast; 1 step is 2 seconds**
- change tab to library
- add one song via "Add + play"
- change tab to playlist
**-> playlist don't show new song until reload**

regards, Andy

**Comment by rern** [comment]
Hi Andy,

- Elapsed counter - Unable to reproduce.
- Add+Play - Should be fixed.

Please try the UPDATE again.

**Comment by AndyFx12** [comment]
Hi rern,

Add+play works fine, now. Thanks!
Elapsed Counter, see here:

https://github.com/user-attachments/assets/5ce28771-c967-4342-9cd6-25a9ea7ebec7

regards, Andy

**Reply by AndyFx12** [reply]
Hi rern,

after some digging into to the code:
- there are two intervals which will be started on tapping to a different playlist-item or click on previous/next track button:
`V.interval.elapsedpl` (at setPlaylistScroll())
and
`V.interval.elapsed` (at setProgressElapsed())

Both execute `S.elapsed++`.

EDIT: I've added a check at setPlaylistScroll, to only increment if `V.interval.elapsed` is not running:
`if ( V.interval.elapsed == 0 ) S.elapsed++;`
and at intervalClear(), I had to add  `V.interval.elapsed=0;` to set them to 0 after clearing.

This workaround seems to work.

regards, Andy

**Reply by rern** [reply]
Hi Andy,

Please try:
```sh
sed -i 's/clearInterval( V.interval.elapsed )/intervalElapsedClear()/' /srv/http/assets/js/function.js

/srv/http/bash/cmd.sh cachebust
```

**Reply by AndyFx12** [reply]
Hi rern,

thanks for your patch - but after this, elapsed counter at playlist is hidden on changing track.

regards, Andy

**Reply by rern** [reply]
Please try the UPDATE again.

**Reply by AndyFx12** [reply]
Hi rern, 

sorry - but update works like your patch. Elapsed counter on playlist of primary device is invisible after changing a track at playlist by tapping an item in the list.

regards, Andy

**Reply by rern** [reply]
Please try the UPDATE again.

**Reply by AndyFx12** [reply]
Hi rern,

it's MUCH better. "Normal" tracks work fine, now. 

Webradio's at playlist have two issues, though:
- because of infinite end-time, there's a useless "/" after elapsed counter
![grafik](https://github.com/user-attachments/assets/e7b10478-9041-4e31-83b0-2b04f2056f34)
- and counter starts but it seems to be stopped at once. Counter seems to be refreshed, if coverart is changing. But most of the time, I can see a stopped counter.

**Reply by AndyFx12** [reply]
Hi rern,

latest Update has fixed this issue as well.

Thanks a lot!
